collect_app/src/main/java/org/odk/collect/android/preferences/ResetDialogPreference.java

```diff
@@ -181,24 +181,7 @@ private void handleResult(List<Integer> resetActions, List<Integer> failedResetA
     }
 
     private void showResultDialog(final String resultMessage) {
-        ((AdminPreferencesActivity) context).runOnUiThread(new Runnable() {
-            @Override
-            public void run() {
-                AlertDialog.Builder b = new AlertDialog.Builder(getContext());
-                b.setTitle(getContext().getString(R.string.reset_app_state_result));
-                b.setMessage(resultMessage);
-                b.setCancelable(false);
-                b.setIcon(android.R.drawable.ic_dialog_info);
-                b.setPositiveButton(getContext().getString(R.string.ok), new DialogInterface.OnClickListener() {
-                    @Override
-                    public void onClick(DialogInterface dialog, int which) {
-                        dialog.dismiss();
-                        ((AdminPreferencesActivity) context).recreate();
-                    }
-                });
-                AlertDialog alertDialog = b.create();
-                alertDialog.show();
-            }
-        });
+        SimpleDialog simpleDialog = SimpleDialog.newInstance(getContext().getString(R.string.reset_app_state_result), android.R.drawable.ic_dialog_info, resultMessage, getContext().getString(R.string.ok));
+        simpleDialog.show(((AdminPreferencesActivity) context).getSupportFragmentManager(), SimpleDialog.COLLECT_DIALOG_TAG);
```

Comment:
Is there a reason you're using `getContext()` in the previous line, but the bare `context` field here? Saving references to a `context` like this class does makes me nervous, as in the past I've run into problems doing so. I'd prefer relying on the `getContext()` method in this line if it works.

Do you think it's safe to cast to `android.support.v4.app.FragmentActivity` rather than `AdminPreferencesActivity`? My intuition is to cast as high as possible in the hierarchy, but I don't have any good motivation for that. I'd be curious to hear what you think.

And what do you think of wrapping this cast in a `try/catch` like [Android does in their dialog docs](https://developer.android.com/guide/topics/ui/dialogs.html#PassingEvents)? If this fails it can just log an error maybe, rather than throw an exception. I don't know that it'd be worth crashing the app over if the wrong `Context` was used. I'm not sure about this one either, it just seems dangerous to trust a cast when contexts are involved.